modules/nf-core/blast/blastp/main.nf

```diff
@@ -23,19 +26,30 @@ process BLAST_BLASTP {
     def prefix = task.ext.prefix ?: "${meta.id}"
     def is_compressed = fasta.name.endsWith(".gz")
     def fasta_name = fasta.name.replace(".gz", "")
+    switch ( out_ext ) {
+        case "xml": outfmt = 5; break
+        case "tsv": outfmt = 6; break
+        case "csv": outfmt = 10; break
+        default:
+            outfmt = '6';
+            out_ext = 'tsv';
+            log.warn("Unknown output file format provided (${out_ext}): selecting BLAST default of tabular BLAST output (tsv)");
+            break
+    }
+
     """
-    if [ "$is_compressed" == "true" ]; then
-        gzip -c -d $fasta > $fasta_name
+    if [ "${is_compressed}" == "true" ]; then
+        gzip -c -d ${fasta} > ${fasta_name}
     fi
```

Comment:
Nvm I guess this is where you got it from. I'll make an issue to address it, not in the scope here 🙃